build-parent/pom.xml

```diff
@@ -89,13 +89,13 @@
         <opensearch.protocol>http</opensearch.protocol>
 
         <!-- Database images for JDBC/Reactive/Hibernate tests and devservices -->
-        <postgres.image>docker.io/postgres:14.2</postgres.image>
-        <mariadb.image>docker.io/mariadb:10.7</mariadb.image>
+        <postgres.image>docker.io/postgres:14</postgres.image>
+        <mariadb.image>docker.io/mariadb:10.6</mariadb.image>
         <db2.image>docker.io/ibmcom/db2:11.5.7.0a</db2.image>
-        <mssql.image>mcr.microsoft.com/mssql/server:2019-CU15-ubuntu-20.04</mssql.image>
+        <mssql.image>mcr.microsoft.com/mssql/server:2019-latest</mssql.image>
         <mysql.image>docker.io/mysql:8.0</mysql.image>
         <oracle.image>docker.io/gvenzl/oracle-xe:21-slim</oracle.image>
-        <mongo.image>docker.io/mongo:4.4.13</mongo.image>
+        <mongo.image>docker.io/mongo:4.4</mongo.image>
```

Comment:
Mongo 5.0.9 is out and 6 is on the horizon.  Shouldn't we bump up to that?

Reply:
> Mongo 5.0.9 is out and 6 is on the horizon. Shouldn't we bump up to that?

Sure - I don't know; @famod asked the same in the comments above, I suppose it's your choice?

Specifically about this line of the patch though - in general I'm not a fan of letting it update versions automatically; in fact I'd prefer we used exact hash ids instead as tests need to be fully reproducible.

Reply:
> Specifically about this line of the patch though - in general I'm not a fan of letting it update versions automatically; in fact I'd prefer we used exact hash ids instead as tests need to be fully reproducible.

Well, it's the same for the other changes and even for the other unchanged versions. (except for db2 which doesn't provide more coarse-grained tags)
It's a tradeoff: Is it more imoportant to have a fully pinned version or is it more important to not ship outdated versions.
I think patch releases should only very rarely cause issues. If they do (e.g. in CI), you can always try a more fine-grained tag to see whether that's it.
We had severly outdated images before and no one really noticed, so IMO we should be more liberal.

Btw, from https://www.postgresql.org/support/versioning/:
> We always recommend that all users run the latest available minor release for whatever major version is in use.

Reply:
As for a major MongoDB upgrade, let's please do it in another PR as the client might need some adjustments.

Reply:
> We had severly outdated images before and no one really noticed, so IMO we should be more liberal.

TBH I did notice, but I'm also totally fine us testing with slightly out of date versions. I don't see the problem with that?

But I have a problem with unreliable integration tests, especially on large projects as there's a lot of people impacted, and many won't be thinking about try a different image version.

There's also an aspect of supply chain attacks here; using an undefined version of anything makes it easier to exploit vulnerabilities.